mautic/mautic
# Dnc report

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 1
| New feature? | 1
| Automated tests included? | 0
| Related user documentation PR URL | 0
| Related developer documentation PR URL | 0
| Issues addressed (#s or URLs) | 0
| BC breaks? | 0
| Deprecations? | 0

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:

This PR adds new **Contacts / Do Not Contact** source to the reports and enables to build reports with the DNC columns.

<img width="1199" alt="screenshot 2018-12-12 at 10 39 56" src="https://user-images.githubusercontent.com/1235442/49861011-3ace0900-fdfb-11e8-9c0d-eaa8025830c2.png">

This PR also fixes a bug introduced in https://github.com/mautic/mautic/pull/6728 where a `1` was being added to the column aliases and some code in `onReportDisplay` methods was expecting the alias without the number.

#### Steps to test this PR:
1. Create new report with the Do Not Contact source.
2. Add all or some DNC columns.
3. Save and see that the data are shown and are correct.
4. Try to filter and group by some DNC columns to ensure these features work.
